sprint_9-solution-http-api

AndreyAA-code opened the pull request.

AndreyAA-code committed:
http classes added

AndreyAA-code committed:
add endpoint

AndreyAA-code committed:
add endpoint

AndreyAA-code committed:
http classes added

AndreyAA-code committed:
http classes added

AndreyAA-code committed:
handler classes added

AndreyAA-code committed:
handler classes added

AndreyAA-code committed:
handler classes added
404 error added

AndreyAA-code committed:
Merge remote-tracking branch 'origin/sprint_9-solution-http-api' into sprint_9-solution-http-api

AndreyAA-code committed:
handler classes added
404 error added

AndreyAA-code committed:
handler classes added
404 error added
epics handler added

AndreyAA-code committed:
handler classes added
404 error added
epics handler added

AndreyAA-code committed:
handler classes added
404 error added
epics handler added

AndreyAA-code committed:
handler classes added
404 error added
epics handler added

AndreyAA-code committed:
add SubtasksHandler

AndreyAA-code committed:
add history handler
add prioritised handler

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

AndreyAA-code committed:
handler classes added
404 error added
epics handler added
http task ready
add checks for path
minor errors checked
http server test created
finish finish
git check

slavrikk commented on src/BaseHttpHandler.java:
```diff
@@ -0,0 +1,48 @@
+import java.time.LocalDateTime;
+
+public class BaseHttpHandler implements HttpHandler {
+    String method;
```
Следует указать модификаторы доступа

AndreyAA-code replied:
выполнено

slavrikk commented on src/BaseHttpHandler.java:
```diff
@@ -0,0 +1,48 @@
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+public class BaseHttpHandler implements HttpHandler {
```
Можно класс обозначить как `abstract` и не имплементировать метод `handle`

AndreyAA-code replied:
выполнено

slavrikk commented on src/BaseHttpHandler.java:
```diff
@@ -0,0 +1,48 @@
+        this.manager = manager;
+    }
+
+    Gson gson = new GsonBuilder()
```
Переменные пишутся перед конструктором

AndreyAA-code replied:
выполнено + предыдущее замечание - классы разнес по пакетам

slavrikk commented on src/BaseHttpHandler.java:
```diff
@@ -0,0 +1,48 @@
+    }
+}
+
+//class EpicTypeToken extends TypeToken<Epic> { //Токен для перевода из json в Epic
```
Лучше удалять закомментированный код

AndreyAA-code replied:
выполнено

slavrikk commented on src/Epic.java:
```diff
@@ -6,16 +6,16 @@
     private List<Integer> epicSubtasks;
-    private Duration duration;
-    private LocalDateTime startTime;
+   // private Duration duration;
```
Лучше удалять закомментированный код

AndreyAA-code replied:
выполнено

slavrikk commented on src/EpicsHandler.java:
```diff
@@ -0,0 +1,57 @@
+import java.io.IOException;
+import java.io.InputStream;
+
+class EpicsHandler extends BaseHttpHandler {
```
Классы лучше обозначать как `public`

slavrikk replied:
Замечание для всего кода

AndreyAA-code replied:
выполнено

slavrikk commented on src/InMemoryTaskManager.java:
```diff
@@ -272,4 +276,29 @@ public List<Task> getPrioritizedTasks() {
     }
+
+    @Override
+    public boolean ifTaskExists(Integer taskId) {
```
Лучше так:

```
    public boolean isTaskExists(Integer taskId) {
      return tasks.containsKey(taskId);
    }
```

slavrikk replied:
Замечание для аналогичных методов

AndreyAA-code replied:
выполнено

slavrikk commented on test/HttpServerTest.java:
```diff
@@ -0,0 +1,303 @@
+    HttpServer httpServer;
+
+    @BeforeEach
+    public void SetUpServer() throws IOException, InterruptedException {
```
Имена методов начинаются с маленькой буквы

AndreyAA-code replied:
выполнено

slavrikk commented on test/HttpServerTest.java:
```diff
@@ -0,0 +1,303 @@
+    }
+
+    @AfterEach
+    public void TearDownServer() {
```
Имена методов начинаются с маленькой буквы

AndreyAA-code replied:
выполнено

AndreyAA-code replied:
выполнено

slavrikk commented on test/HttpServerTest.java:
```diff
@@ -0,0 +1,303 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+public class HttpServerTest {
+    final TaskManager manager = new InMemoryTaskManager();
```
Новый объект manager лучше создавать в методе `@BeforeEach`, тогда можно не удалять таски в методе `@AfterEach`

AndreyAA-code replied:
выполнено

slavrikk reviewed (commented):
Так же хорошо бы все классы разнести по пакетам, чтобы они не лежали все в src

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add

AndreyAA-code committed:
add history handler
add prioritised handler
add 406 code for update
check epic add
packages add
all suggestions applied
